lib/AST/NameLookup.cpp

```diff
@@ -1791,9 +1791,10 @@ resolveTypeDeclsToNominal(Evaluator &evaluator,
         }
 
         // Type version: an empty class-bound existential.
-        if (auto type = typealias->getUnderlyingTypeLoc().getType()) {
-          if (type->isAnyObject())
-            anyObject = true;
+        if (typealias->hasInterfaceType()) {
```

Comment:
The hasInterfaceType() check should not be necessary?

Reply:
Requesting the underlying type potentially calls back into validation.  This is also justified because the interface type separation changes mean that `AnyObject` declarations will always have their interface type set by the time we see them.